### ISSUE TITLE
Associate Venues with Chapters

### ISSUE THREAD

**Issue by ojeytonwilliams** [freeCodeCamp/chapter#1195]
Do we want each venue to belong to one chapter? There are couple of reasons I can see to do this, one practical and one technical.

The practical reason is that a venue is typically going to be close to the chapter members and so forcing event creators to choose from all possible venues is unhelpful.  There is also the potential for confusion if admins of multiple chapters update the same venue.

The technical reason is that the authorisation logic uses the target `chapterId` to determine if a user's request should be rejected.  If `chapterId` is not sent as part of the request, then chapter admins will not be able to manage venues.  There are a few potential workarounds: 

1) Give chapter admins the VenuesManage instance role
2) Handle the authorization logic in the resolver
3) Rework authorization logic

Issues:

1) is pretty awkward. In part because it's messy (users shouldn't need more than one role) and in part because instance permissions cannot be stopped by bans (yet).
2) mixes up concerns
3) is the nuclear option, since I'm not sure what we'd do instead

## Summing up

Given that there are positive reasons to want this as well as a lack of obvious workarounds, I think this is the way to go.

Any concerns before I get on with implementing this?

**Comment by allella** [comment]
I guess what you're asking is if there would be global venues, or else duplicate venues in situations where two or more Chapters meet at the same location.

The use case I have isn't the base case, but having global venues wouldn't work well for it.  That is, I plan to run an instance of Chapter for my local area as a resource for all of the small meetup groups that don't / won't have a parent organization hosting an instance for them.  So, in that case we'd have many events in the same venue for different _organizers_.

I think making the venues specific to the Chapter could work. The database would possibly have duplicate venues, but IF you're filtering the list of available venues to only showing the venues that Chapter admin / organizer created, then nobody would see the duplicates and it wouldn't be confusing because nobody would see venues they didn't create.

**Comment by allella** [comment]
I guess a middle-ground on global vs duplicate venues is if you had a flag on the database where some venues could be set as global, and therefore available to all _events_?

I'm suggesting this more as a post-MVP, but it could give some flexibility to where an owner or admin could control the list of approved venues. Or, the owner / admin could allow organizers to maintain their own subset of venues. Or, allow for a mix of the two.

**Comment by ojeytonwilliams** [comment]
Maybe a safe option would just be the ability to view a read-only list of other chapter's venues.

So, the user story would be something like this:

- As a Chapter admin, I can see a list of all Venues and import copies of them into my Chapter

This way it's clear who has the right to modify things (admins of the appropriate chapters and owners), but we cut down on the busy-work of setting up a venue.

**Comment by allella** [comment]
Even with the standard use case, I could see the venue list getting messy if organizers are left to their own devices.

If someone is going to see a "standardized" list of venues, then it may be best that only an admin has that ability so you don't end up with organizers doing things different ways and creating noise for each other.

For example, you might end up with organizers creating a new, effectively duplicate, _venue_ just to differentiate a floor or area or room. For instance location, like "The Spot - 1st Floor" vs "The Spot - 3rd Floor, Room 3".

Having a standardized list of venues with GPS coordinates would be nice to show on a map and such, but it seems like Meetup and Eventbrite don't attempt to offer venue suggestions because it's unlikely to fit the long-tail of things organizers want or expect the venue to do.

I feel like having a "global" / "universal" (or some better name) flag on the events could offer the best of both worlds if the permissions, post-MVP, allowed for the the admin / owner to define who can create a venue.

Only Admins Can Create Venues - admins create all venues and mark them as global and organizers can pick from the defined / approved list
Organizers Can Create Own Venues - organizers can create venues and can only see the ones they've created.
Mixed - Organizers can see a list that combines (global + own), but not non-global venues created by other organizers.

**Comment by ojeytonwilliams** [comment]
Sounds good to me.  Once organisers exist, they will definitely need to be able to create venues, even if they mostly use existing ones.

I suspect we'll only truly know once we've used it for a while, but the Mixed approach seems good. It gives organisers all the freedom they'll need while still saving them time (assuming they check the global list).

**Comment by ojeytonwilliams** [comment]
Back to the MVP: having more than one type of list of venues isn't minimal and if we're picking one, chapter specific lists seem best.

**Comment by allella** [comment]
Sure, if you can start with authorization for chapter specific and it won't complicate something like adding a global venues flag, then that sounds like a decent MVP.

**Comment by ojeytonwilliams** [comment]
Reopening since there is more to be done.  Context: https://github.com/freeCodeCamp/chapter/pull/1212#issuecomment-1184301773

**Comment by ojeytonwilliams** [comment]
You're more than welcome to work on this.  My next task is move us from using magic links for Authorization to using Auth0, so I won't be touching much else until that's done.
